pkg/ring/kv/client.go

```diff
@@ -67,9 +67,17 @@ func (cfg *Config) RegisterFlagsWithPrefix(flagsPrefix, defaultPrefix string, f
 // It also deals with serialisation by using a Codec and having a instance of
 // the the desired type passed in to methods ala json.Unmarshal.
 type Client interface {
+	// List returns a list of keys under the given prefix. The key names will
+	// include the prefix name.
+	List(ctx context.Context, prefix string) ([]string, error)
+
 	// Get a specific key.  Will use a codec to deserialise key to appropriate type.
+	// If the key does not exist, Get will return nil and no error.
 	Get(ctx context.Context, key string) (interface{}, error)
 
+	// Delete a specific key. Returns true if an object was deleted.
```

Comment:
Could you also document the case the key doesn't exist (what is expected to return?) just to write the expected behaviour on stone for future client implementations :)

Reply:
Consul API doesn't provide this feedback... it returns 200 OK even if key didn't exist before DELETE. Consul client tries to handle that by first Getting the value and then using DeleteCAS, which can fail if value has changed in the meantime -- in a totally unexpected way, based on this method signature.

I would suggest to remove `bool` return value, and make Delete best-effort. If it returns no error, client can assume that key was deleted, but it cannot assume that key is no longer in the KV store, since another client might have written it again. I think it would be better than returning errors in such case (as consul client does can do right now).

Reply:
That's a good point. I'll change it to do this and do a `Get` before `Delete` from the agent side (even though it's subject to race conditions, it's unlikely to be a problem for the Agent's use case)